src/audio/coreaudio/SDL_coreaudio.m

```diff
@@ -1145,6 +1145,129 @@ output device (in which case we'll try again). */
     return (this->hidden->thread != NULL) ? 0 : -1;
 }
 
+static int
+COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
+{
+    UInt32 alive = 0;
+    pid_t pid = 0;
+
+    AudioObjectPropertyAddress addr = {
+        iscapture ? kAudioHardwarePropertyDefaultInputDevice
+                  : kAudioHardwarePropertyDefaultOutputDevice,
+        iscapture ? kAudioDevicePropertyScopeInput
+                  : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+        
+    AudioObjectPropertyAddress nameaddr = {
+        kAudioObjectPropertyName,
+        iscapture ? kAudioDevicePropertyScopeInput
+                  : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+    
+    AudioObjectPropertyAddress freqaddr = {
+        kAudioDevicePropertyNominalSampleRate,
+        iscapture ? kAudioDevicePropertyScopeInput
+                  : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+    
+    AudioObjectPropertyAddress bufaddr = {
+        kAudioDevicePropertyStreamConfiguration,
+        iscapture ? kAudioDevicePropertyScopeInput : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+    
+    AudioDeviceID devid;
+    OSStatus result = noErr;
+    UInt32 size = 0;
+    
+    // Get the Device ID
+    size = sizeof (AudioDeviceID);
+    result = AudioObjectGetPropertyData(kAudioObjectSystemObject, &addr,
+                                        0, NULL, &size, &devid);
+    
+    if (result != noErr) {
+        return SDL_SetError("%s: Default Device ID not found", "coreaudio");
+    }
+    
+    
+    // Use the Device ID to get the name
+    CFStringRef cfstr = NULL;
```

Comment:
A little more move of locals needed (for clang >= 14 as explained in PR/5854)
Also remove some unnecessary initializations and unused locals. Patch attached
here: [patch.txt](https://github.com/libsdl-org/SDL/files/9637413/patch.txt), also inlined below.
```diff
diff --git a/src/audio/coreaudio/SDL_coreaudio.m b/src/audio/coreaudio/SDL_coreaudio.m
index a3ca542..346f180 100644
--- a/src/audio/coreaudio/SDL_coreaudio.m
+++ b/src/audio/coreaudio/SDL_coreaudio.m
@@ -1148,9 +1148,15 @@ COREAUDIO_OpenDevice(_THIS, const char *devname)
 static int
 COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
 {
-    UInt32 alive = 0;
-    pid_t pid = 0;
-
+    AudioDeviceID devid;
+    AudioBufferList *buflist;
+    OSStatus result;
+    UInt32 size;
+    CFStringRef cfstr;
+    char *devname;
+    int usable;
+    double sampleRate;
+    
     AudioObjectPropertyAddress addr = {
         iscapture ? kAudioHardwarePropertyDefaultInputDevice
                   : kAudioHardwarePropertyDefaultOutputDevice,
@@ -1158,7 +1164,7 @@ COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
                   : kAudioDevicePropertyScopeOutput,
         kAudioObjectPropertyElementMaster
     };
-        
+    
     AudioObjectPropertyAddress nameaddr = {
         kAudioObjectPropertyName,
         iscapture ? kAudioDevicePropertyScopeInput
@@ -1179,10 +1185,6 @@ COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
         kAudioObjectPropertyElementMaster
     };
     
-    AudioDeviceID devid;
-    OSStatus result = noErr;
-    UInt32 size = 0;
-    
     // Get the Device ID
     size = sizeof (AudioDeviceID);
     result = AudioObjectGetPropertyData(kAudioObjectSystemObject, &addr,
@@ -1192,9 +1194,8 @@ COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
         return SDL_SetError("%s: Default Device ID not found", "coreaudio");
     }
     
-    
     // Use the Device ID to get the name
-    CFStringRef cfstr = NULL;
+    cfstr = NULL;
     size = sizeof (CFStringRef);
     result = AudioObjectGetPropertyData(devid, &nameaddr, 0, NULL, &size, &cfstr);
     
@@ -1205,8 +1206,8 @@ COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
     CFIndex len = CFStringGetMaximumSizeForEncoding(CFStringGetLength(cfstr),
                                                     kCFStringEncodingUTF8);
     
-    char *devname = (char *) SDL_malloc(len + 1);
-    int usable = ((devname != NULL) &&
+    devname = (char *) SDL_malloc(len + 1);
+    usable = ((devname != NULL) &&
               (CFStringGetCString(cfstr, devname, len + 1, kCFStringEncodingUTF8)));
     CFRelease(cfstr);
     
@@ -1231,7 +1232,7 @@ COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
     // Uses the Device ID to get the spec
     SDL_zero(*spec);
 
-    double sampleRate = 0;
+    sampleRate = 0;
     size = sizeof(sampleRate);
     result = AudioObjectGetPropertyData(devid, &freqaddr, 0, NULL, &size, &sampleRate);
 
@@ -1245,7 +1246,7 @@ COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
     if (result != noErr)
         return SDL_SetError("%s: Default Device Data Size not found", "coreaudio");
 
-    AudioBufferList *buflist = (AudioBufferList *) SDL_malloc(size);
+    buflist = (AudioBufferList *) SDL_malloc(size);
     if (buflist == NULL)
         return SDL_SetError("%s: Default Device Buffer List not found", "coreaudio");
 
diff --git a/test/testsurround.c b/test/testsurround.c
index 7761817..ff55af7 100644
--- a/test/testsurround.c
+++ b/test/testsurround.c
@@ -138,6 +138,7 @@ main(int argc, char *argv[])
     char *deviceName;
     SDL_AudioSpec spec;
     int i;
+
     /* Enable standard application logging */
     SDL_LogSetPriority(SDL_LOG_CATEGORY_APPLICATION, SDL_LOG_PRIORITY_INFO);
 
```

Reply:
OK, I don't have anything else.

iOS / tvOS build failures still need resolving.